src/js/components/TimelineUI.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import TweetBlock from "./TweetBlock";
+import {fetchHomeTimeline} from "../twitter-api.js";
+import "../../css/components/TimelineUI.scss";
+
+// Presentational Component for Timeline
+export default class TimelineUI extends React.Component {
+
+	constructor() {
+		super();
+		this.state = {
+			tweets: null,
+			error: null
+		}
+	}
+
+	componentDidMount() {
+		this.updateTimeline();
+	}
+
+	render() {
+		let displayedElem; // Either display error or tweets
+		if (this.state.error) {
+			displayedElem = <div id="error-div">{this.state.error}</div>
+		}
+		else if (this.state.tweets) {
+			displayedElem = 
+				<div id="tweets">{
+					this.state.tweets.map((tweet, index) => 
+						<div key={tweet.url} className="row">
+							<TweetBlock tweet={tweet}/>
+						</div>
+					)
+				}
+			</div>;
+		} 
+		return (
+		    <div id="timeline-div">
+				<button id="update-timeline" onClick={this.updateTimeline.bind(this)}>Update Home Timeline</button>
+				{displayedElem}
+			</div>
+		);
+	}
+
+	updateTimeline() {
+		fetchHomeTimeline().then(response => {
+			this.setState(response);
```

Comment:
what is response? what if response has more k:v pairs than you expected?

Reply:
The response here is an array of tweet objects. Is the concern that I could be storing data to state that I'm not using if the tweet objects have extra k:v pairs?

Reply:
yep, do you want to be setting state in your timelineUI component for things it may potentially not care about?